quic: multiple refactors (#207)
nodejs/quic: doc/api/quic.md

});
```

## QUIC Basics

> QUIC is a UDP-based network transport protocol that includes built in security

Reviewer: ```suggestion
QUIC is a UDP-based network transport protocol that includes built-in security
```

These will also be available using the `quicsession.servername`,
`quicsession.alpnProtocol`, and `quicsession.cipher` properties.

#### Event: `'stream'`
<!-- YAML
added: REPLACEME
-->

Emitted when a new `QuicStream` has been initiated by the connected peer.

#### quicsession.ackDelayRetransmitCount
<!-- YAML
added: REPLACEME
-->

* Type: {BigInt}

> A `BigInt` representing the number of retranmissions caused by delayed

Reviewer: ```suggestion
A `BigInt` representing the number of retransmissions caused by delayed
```

The ALPN protocol identifier negotiated for this session.

#### quicsession.authenticated
<!--YAML
added: REPLACEME
-->
* Type: {boolean}

True if the certificate provided by the peer during the TLS 1.3
handshake has been verified.

#### quicsession.authenticationError

* Type: {Object} An error object

If `quicsession.authenticated` is false, returns an `Error` object
representing the reason the peer certificate verification failed.

#### quicsession.bidiStreamCount
<!-- YAML
added: REPLACEME
-->

* Type: {BigInt}

A `BigInt` representing the total number of bidirectional streams
created for this `QuicSession`.

#### quicsession.bytesInFlight
<!-- YAML
added: REPLACEME
-->

* Type: {number}

> The total number of unacknoledged bytes this QUIC endpoint has tramitted

Reviewer: ```suggestion
The total number of unacknowledged bytes this QUIC endpoint has transmitted
```

Set to `true` if the TLS handshake has completed.

#### quicsession.handshakeDuration
<!-- YAML
added: REPLACEME
-->

* Type: {BigInt}

A `BigInt` representing the length of time taken to complete the TLS handshake.

#### quicsession.keyUpdateCount
<!-- YAML
added: REPLACEME
-->

* Type: {BigInt}

A `BigInt` representing the number of key update operations that have
occured.

#### quicsession.latestRTT
<!-- YAML
added: REPLACEME
-->

* Type: {BigInt}

The most recently recorded RTT for this `QuicSession`.

#### quicsession.lossRetransmitCount
<!-- YAML
added: REPLACEME
-->

* Type: {BigInt}

> A `BigInt` representing the number of lost-packet retranmissions that have been

Reviewer: ```suggestion
A `BigInt` representing the number of lost-packet retransmissions that have been
```

performed on this `QuicSession`.

#### quicsession.maxDataLeft
<!-- YAML
added: REPLACEME
-->

* Type: {Number}

The total number of bytes the `QuicSession` is *currently* allowed to
send to the connected peer.

#### quicsession.maxInFlightBytes
<!-- YAML
added: REPLACEME
-->

* Type: {BigInt}

> A `BigInt` representing the maximum number of in flight bytes recorded

Reviewer: ```suggestion
A `BigInt` representing the maximum number of in-flight bytes recorded
```

operations. There is no return value and there is no way to monitor the status
of the `ping()` operation.

#### quicsession.peerInitiatedStreamCount
<!-- YAML
added: REPLACEME
-->

* Type: {BigInt}

> A `BigInt` representing the total number of `QuicStreams` initated by the

Reviewer: ```suggestion
A `BigInt` representing the total number of `QuicStreams` initiated by the
```

The callback *must* be invoked in order for the TLS handshake to continue.

#### quicserversession.addContext(servername[, context])
<!-- YAML
added: REPLACEME
-->

* `servername` {String} A DNS name to associate with the given context.

> * `context` {Object} A TLS SecureContext to associate with the `servername`.

Reviewer: ```suggestion
* `context` {tls.SecureContext} A TLS SecureContext to associate with the `servername`.
```

});
```

#### Event: `'readable'`
<!-- YAML
added: REPLACEME
-->

#### quicstream.aborted
<!-- YAML
added: REPLACEME
-->
* Type: {boolean}

> True if dataflow on the `QuicStream` was prematured terminated.

Reviewer: ```suggestion
True if dataflow on the `QuicStream` was prematurely terminated.
```
